crates/arrow_tools/src/schema.rs

```diff
@@ -91,3 +93,46 @@ pub fn expand_views_schema(schema: &Schema) -> Schema {
 
     Schema::new(transformed_fields)
 }
+
+pub fn set_computed_columns_meta<S: ::std::hash::BuildHasher>(
+    schema: &mut Schema,
+    computed_columns_meta: &HashMap<String, Vec<String>, S>,
+) {
+    for (base_column, computed_columns) in computed_columns_meta {
+        set_computed_columns_meta_for_base_column(schema, base_column, computed_columns);
+    }
+}
+
+pub fn set_computed_columns_meta_for_base_column(
+    schema: &mut Schema,
+    base_column: &str,
+    computed_columns: &[String],
+) {
+    schema.metadata.insert(
```

Comment:
An alternative approach considered was using field metadata directly, e.g., adding an `"is_computed": "base_col_name"` key-value pair. However, fields cannot be mutated as they are `Arc` references and do not implement `DerefMut`.